agent/xds/envoy_versioning.go

```diff
@@ -27,24 +27,15 @@ type unsupportedVersion struct {
 
 type supportedProxyFeatures struct {
 	// add version dependent feature flags here
-
-	// GatewaysNeedStubClusterWhenEmptyWithIncrementalXDS is needed to paper
-	// over some weird envoy behavior.
 	//
-	// For some reason Envoy versions prior to 1.16.0 when sent an empty CDS
-	// list via the incremental xDS protocol will correctly ack the message and
-	// just never request LDS resources.
-	GatewaysNeedStubClusterWhenEmptyWithIncrementalXDS bool
-
-	// IncrementalXDSUpdatesMustBeSerial is needed to avoid an envoy crash.
+	// For example, we previously had flags for Envoy < 1.16 called:
```

Comment:
I like this. Prior to these 2 flags existing the struct was empty for a bit. Prior to that it had completely different features available. When I went to add these two flags I had to look at my own file history to see how to re-add a feature flag.